packages/ui/components/button/Button.tsx

```diff
@@ -33,38 +22,125 @@ export type ButtonBaseProps = {
   flex?: boolean;
 };
 export type ButtonProps = ButtonBaseProps &
+  VariantProps<typeof buttonClasses> &
   (
     | (Omit<JSX.IntrinsicElements["a"], "href" | "onClick" | "ref"> & LinkProps)
     | (Omit<JSX.IntrinsicElements["button"], "onClick" | "ref"> & { href?: never })
   );
 
-const variantClassName = {
-  primary:
-    "border border-transparent text-white bg-brand-500 hover:bg-brand-400 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-brand-500",
-  secondary: "border border-gray-200 text-brand-900 bg-white hover:bg-gray-100",
-  minimal:
-    "text-gray-700 bg-transparent hover:bg-gray-100 focus:outline-none focus:ring-offset-1 focus:bg-gray-100 focus:ring-brand-900 dark:text-darkgray-900 hover:dark:text-gray-50",
-  minimalSecondary:
-    "text-gray-700 bg-transparent hover:bg-gray-100 dark:hover:bg-darkgray-200 focus:outline-none focus:ring-2 focus:ring-offset-1 focus:bg-gray-100 focus:ring-brand-900 dark:text-darkgray-900 hover:dark:text-gray-50 border border-transparent hover:border-gray-300 dark:hover:border-darkgray-300",
-  destructive:
-    "text-gray-900 focus:text-red-700 bg-transparent hover:bg-red-100 hover:text-red-700 focus:outline-none focus:ring-2 focus:ring-offset-1 focus:bg-red-100 focus:ring-red-700",
-};
-const variantDisabledClassName = {
-  primary: "border border-transparent bg-brand-500 bg-opacity-20 text-white",
-  secondary: "border border-gray-200 text-brand-900 bg-white opacity-30",
-  minimal: "text-gray-400 bg-transparent",
-  minimalSecondary: "text-gray-400 bg-transparent",
-  destructive: "text-red-700 bg-transparent opacity-30",
-};
+const buttonClasses = cva(
+  "inline-flex items-center text-sm font-medium relative rounded-md transition-colors",
+  {
+    variants: {
+      color: {
+        primary: "text-white dark:text-black",
+        secondary: "text-gray-900 dark:text-darkgray-900",
+        minimal: "text-gray-900 dark:text-darkgray-900",
+        destructive: "",
+      },
+      size: {
+        base: "h-9 px-4 py-2.5  ",
+        lg: "h-[36px] px-4 py-2.5 ",
+        icon: "flex justify-center min-h-[36px] min-w-[36px] ",
+      },
+      loading: {
+        true: "cursor-wait",
+      },
+      disabled: {
+        true: "cursor-not-allowed",
+      },
+    },
+    compoundVariants: [
```

Comment:
The compound variants sometimes look a bit much, but I have a few things to say about that: 
* There's actually more styles now than there was before I switched to CVA, so it really is more. Needed to look like design. 
* There's less possible bugs now, since before we simple concatenated disabled states after the regular states (eg 2 background colors), and totally relied on the order in which the CSS ended up in de CSS file for the correct styles to apply. Now there's no duplicate styles anymore. 
* This makes it WAY easier in my opinion to see which styles are applied, compared to the concatenations spread throughout the file before.